services/QuillLMS/client/app/bundles/Shared/components/shared/tooltip.tsx

```diff
@@ -27,42 +28,43 @@ export class Tooltip extends React.Component<TooltipProps, {}> {
     this.hideTooltip = this.hideTooltip.bind(this)
   }
 
-
-  componentWillUnmount() {
-    document.removeEventListener('click', this.hideTooltipOnClick, false)
-  }
-
-  hideTooltipOnClick(e) {
-    const { handleClick } = this.props;
-    if (this.tooltip && e.target !== this.tooltip && e.target !== this.tooltipTrigger) {
-      this.hideTooltip()
-    }
-    if(handleClick) {
-      handleClick(e);
-    }
-  }
-
   showTooltip() {
     const { tooltipText } = this.props
     const activeTooltips = document.getElementsByClassName('visible quill-tooltip')
     Array.from(activeTooltips).forEach(tooltip => tooltip.classList.remove('visible'))
     clearTimeout(this.timer)
     this.tooltip.innerHTML = tooltipText
     this.tooltip.classList.add('visible')
+    this.setState({ tooltipVisible: true })
   }
 
   hideTooltip() {
     this.tooltip.classList.remove('visible')
     this.tooltip.innerHTML = ''
+    this.setState({ tooltipVisible: false })
   }
 
   startTimer() {
     this.timer = setTimeout(this.hideTooltip, 1500)
   }
 
+  handleTooltipClick = (e) => {
+    const { tooltipVisible } = this.state
+
+    if(tooltipVisible) {
+      this.setState({ tooltipVisible: false })
+      this.hideTooltip()
+    } else {
+      this.setState({ tooltipVisible: true })
+      this.showTooltip()
```

Comment:
Same here - since we call `setState` in `showTooltip` we shouldn't need it here too.